data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -94,91 +106,122 @@ public S3SelectObjectWorker(final S3ObjectRequest s3ObjectRequest) {
      */
     public void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException {
         try{
-            selectObjectContent(s3ObjectReference);
-        }catch(Exception e){
+            selectObjectInBatches(s3ObjectReference);
+        } catch (Exception e){
+            LOG.error("Unable to process object reference: {}", s3ObjectReference, e);
             s3ObjectPluginMetrics.getS3ObjectsFailedCounter().increment();
             throw new IOException(e);
         }
     }
 
-    private void selectObjectContent(S3ObjectReference s3ObjectReference) throws IOException {
-        final InputStream inputStreamList;
-        final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
+    private void selectObjectInBatches(final S3ObjectReference s3ObjectReference) throws IOException {
         final InputSerialization inputSerialization = getInputSerializationFormat(serializationFormatOption);
-        final SelectObjectContentRequest selectObjectContentRequest = getS3SelectObjRequest(s3ObjectReference, inputSerialization);
-        s3AsyncClient.selectObjectContent(selectObjectContentRequest, s3SelectResponseHandler).join();
-        if(s3SelectResponseHandler.getException()!=null)
-            throw new IOException(s3SelectResponseHandler.getException());
-        final List<SelectObjectContentEventStream> receivedEvents = s3SelectResponseHandler.getS3SelectContentEvents();
-        if (!receivedEvents.isEmpty()) {
-            inputStreamList = getInputStreamFromResponseHeader(s3SelectResponseHandler);
-            parseCompleteStreamFromResponseHeader(s3ObjectReference, bufferAccumulator, inputStreamList);
-            s3ObjectPluginMetrics.getS3ObjectEventsSummary().record(bufferAccumulator.getTotalWritten());
-            s3ObjectPluginMetrics.getS3ObjectsSucceededCounter().increment();
-            receivedEvents.clear();
-        }else{
-            LOG.info("S3 Select returned no events for S3 object {}", s3ObjectReference);
+        InputStream inputStreamList;
+        final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
+
+        final long objectSize = getObjectSize(s3ObjectReference);
+
+        long startRange = 0;
+        long endRange = Math.min(MAX_S3_OBJECT_CHUNK_SIZE, objectSize);
+        while (startRange < objectSize) {
+            final ScanRange scanRange = ScanRange.builder()
+                    .start(startRange)
+                    .end(endRange)
+                    .build();
+            final SelectObjectContentRequest selectObjectContentRequest = getS3SelectObjRequest(s3ObjectReference, inputSerialization, scanRange);
+
+            s3AsyncClient.selectObjectContent(selectObjectContentRequest, s3SelectResponseHandler).join();
```

Comment:
#2447 introduced issues with timeouts on larger files. These timeouts went away when reverting the client overrides set here: https://github.com/opensearch-project/data-prepper/pull/2447/files#diff-49e2eeca5ae076b473ca54aed8fb03db3edc2b074180cb59876f50ac72dcd6ffR52-R57

Could you include that revert in this PR? We can un-revert later if needed.

@udaych20 could you elaborate on why the defaults were overridden in that PR? I want to be sure there isn't an underlying issue that I have not yet encountered that necessitated the default timeouts being overridden.

Reply:
@engechas sometimes we are facing below exception while doing testing. so we have overriden the changes. 

2023-03-31T17:46:33,364 [Thread-1] ERROR org.opensearch.dataprepper.plugins.source.ScanObjectWorker - Error while process the parseS3Object. 
java.io.IOException: java.util.concurrent.CompletionException: software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: Acquire operation took longer than the configured maximum time. This indicates that a request cannot get a connection from the pool within the specified maximum time. This can be due to high request rate.

Consider taking any of the following actions to mitigate the issue: increase max connections, increase acquire timeout, or slowing the request rate.

Increasing the max connections can increase client throughput (unless the network interface is already fully utilized), but can eventually start to hit operation system limitations on the number of file descriptors used by the process. If you already are fully utilizing your network interface or cannot further increase your connection count, increasing the acquire timeout gives extra time for requests to acquire a connection before timing out. If the connections doesn't free up, the subsequent requests will still timeout.

If the above mechanisms are not able to fix the issue, try smoothing out your requests so that large traffic bursts cannot overload the client, being more efficient with the number of times you need to call AWS, or by increasing the number of hosts sending requests.